### PR TITLE
[EDS 552] Fix phone number getting chomped during input validation

### PR DESCRIPTION
**Pull request by tomthorogood**
Closes EDS-552

### CODE REVIEW

**Comment by tomthorogood** [comment]
Hmm, one thing I just realized: the output field `query` is now a lie. It should be called `request` instead, since the query is different than the request.

**Comment by goulter** [comment]
> Hmm, one thing I just realized: the output field `query` is now a lie. It should be called `request` instead, since the query is different than the request.
> 
> I'll make this change for both front- and back-ends

Thanks for taking care of both ends. How is the query different from the request? I don't doubt it, but can you clarify?

**Comment by tomthorogood** [comment]
The query to PWS is `""` because we sanitize the input, but the user requested `abcdef`. Therefore, the field being called `query` is misleading, because the actual query to PWS removes all non-digits. Changing the field name to `request` better indicates that the value is what the _user_ requested, not what we submitted to the PWS API.

**Comment by goulter** [comment]
Ah, I see. Makes sense.

**Comment by tomthorogood** [comment]
@goulter updated; this also fixes is a bug in the `pre-push.sh` script.

**Comment by tomthorogood** [comment]
Oops, noticed a formatting issue.

**Comment by tomthorogood** [comment]
Ok, updated again, and I amended the commits to separate the bugfix.

**Comment by goulter** [comment]
Does this mean `query` in the returned `search_results` object is completely replaced by `request`?
In other words, do we need to replace all templates instances of `query` with `request`? 
For example [dynamic_search_options.html](https://github.com/UWIT-IAM/uw-husky-directory/blob/main/husky_directory/templates/dynamic_search_options.html)

**Comment by tomthorogood** [comment]
yes, thanks I missed that file.

**Comment by tomthorogood** [comment]
how bout those front-end tests ;)

**Comment by tomthorogood** [comment]
(that's on me too, i need to make a docker composition that would actually run them)

**Comment by tomthorogood** [comment]
updated

File: husky_directory/templates/search.html [hunk 1]
```diff
@@ -6,7 +6,7 @@ <h3>Search by</h3>
                             <div class="input-group">
                                 <input type="text" class="form-control"
                                        {% if search_result is defined %}
-                                           value="{{ search_result['query'].values()|first }}"
+                                           value="{{ search_result['request'].values()|first }}"
```
Comment (goulter) [inline husky_directory/templates/search.html:9]: [One more](https://github.com/UWIT-IAM/uw-husky-directory/blob/main/husky_directory/templates/search.html#L14) changed needed in this file

Comment (tomthorogood) [reply husky_directory/templates/search.html:9]: I disagree. I think at that state, it is the user's query that they are inputting. It is not yet a request. It's the search field for their query, that we manipulate.

Comment (tomthorogood) [reply husky_directory/templates/search.html:9]: When we return a response, we include the request we received.

Comment (tomthorogood) [reply husky_directory/templates/search.html:9]: (I purposely did not change that one.)

Comment (tomthorogood) [reply husky_directory/templates/search.html:9]: To put it more concisely:

From the context of the front-end, the user is entering a query.
From the context of the back-end, the server is including the original request with the response.

Comment (goulter) [reply husky_directory/templates/search.html:9]: Ok, this should still work with the [js](https://github.com/UWIT-IAM/uw-husky-directory/blob/main/husky_directory/static/js/setSearchNameAttribute.js#L10).

Comment (goulter) [reply husky_directory/templates/search.html:9]: I'll give it a go.

Comment (tomthorogood) [reply husky_directory/templates/search.html:9]: yes, i mentioned i tested this...

Comment (tomthorogood) [reply husky_directory/templates/search.html:9]: (don't forget, you can too!)

**Review by goulter — APPROVED** [review]

**Pull request merged**